src/commands/aksRetinaCapture/aksRetinaCapture.ts

```diff
@@ -0,0 +1,133 @@
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return;
+    }
+
+    if (!cloudExplorer.available) {
+        vscode.window.showWarningMessage(`Cloud explorer is unavailable.`);
+        return;
+    }
+
+    if (!clusterExplorer.available) {
+        vscode.window.showWarningMessage(`Cluster explorer is unavailable.`);
+        return;
+    }
+
+    const clusterInfo = await getKubernetesClusterInfo(target, cloudExplorer, clusterExplorer);
+    if (failed(clusterInfo)) {
+        vscode.window.showErrorMessage(clusterInfo.error);
+        return;
+    }
+
+    const kubectlRetinaPath = await getRetinaBinaryPath();
+    if (failed(kubectlRetinaPath)) {
+        vscode.window.showWarningMessage(`kubectl retina path was not found ${kubectlRetinaPath.error}`);
+        return;
+    }
+
+    ensureDirectoryInPath(path.dirname(kubectlRetinaPath.result));
+
+    const extension = getExtension();
+    if (failed(extension)) {
+        vscode.window.showErrorMessage(extension.error);
+        return;
+    }
+
+    const kubeConfigFile = await tmpfile.createTempFile(clusterInfo.result.kubeconfigYaml, "yaml");
+    console.log(`Kubeconfig file: ${kubeConfigFile}`);
+
+    // Get all Nodes For this Cluster
+    const nodelistResult = await invokeKubectlCommand(
+        kubectl,
+        kubeConfigFile.filePath,
+        `get nodes -o jsonpath="{range .items[*]}{.metadata.name}{' '}{end}"`,
+    );
+
+    if (failed(nodelistResult)) {
+        vscode.window.showErrorMessage(`Failed to capture the cluster: ${nodelistResult.error}`);
+        return;
+    }
+    const nodesList = nodelistResult.result.stdout.trim().split(" ").map((nodeName) => ({ label: nodeName }));
+
+    // Pick a cluster from group to compare with
+    const nodeNamesSelected = vscode.window.showQuickPick(nodesList, {
+        canPickMany: true,
+        placeHolder: "Please select all the Nodes you want Retina to capture traffic from.",
+        title: "Select Nodes to Capture Traffic From",
+    });
+
+    const selectedNodes = await nodeNamesSelected.then((selectedItems) => {
+        const selectedNodes = selectedItems?.map((item) => item.label).join(",") ?? "";
+        return selectedNodes;
+    });
+
+    if (!selectedNodes) {
+        return;
+    }
+
+    // Retina Run Capture
+    const retinaCaptureResult = await longRunning(`Retina Distributed Capture running for cluster ${clusterInfo.result.name}.`, async () => {
+        return await invokeKubectlCommand(
+            kubectl,
+            kubeConfigFile.filePath,
+            `retina capture create --host-path /mnt/capture --node-selectors "kubernetes.io/os=linux" --node-names "${selectedNodes}" --no-wait=false`,
+        )
+    });
+
+    if (failed(retinaCaptureResult)) {
+        vscode.window.showErrorMessage(`Failed to capture the cluster: ${retinaCaptureResult.error}`);
+        return;
+    }
+
+    if (retinaCaptureResult.result.stderr || retinaCaptureResult.result.code !== 0) {
```

Comment:
I don't think there's any need to check `code`, `strerr` or `stdout` here, so lines 100-107 could be removed. It's already covered by the `if (failed(retinaCaptureResult))` check.